src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -573,7 +573,7 @@ private static Long findSchema(Connection connection, Position targetPosition, L
 			// Only consider binlog positions before the target position on the current server.
 			// Within those, sort for the latest binlog file, then the latest binlog position.
 			PreparedStatement s = connection.prepareStatement(
-				"SELECT id from `schemas` "
+				"SELECT id, binlog_position, base_schema_id from `schemas` "
 				+ "WHERE deleted = 0 "
 				+ "AND last_heartbeat_read <= ? AND ((binlog_file < ?) OR (binlog_file = ? and binlog_position <= ?)) AND server_id = ? "
```

Comment:
I think this whole change could actually be boiled down to changing `binlog_position <= ?` to `binlog_position < ?`, couldn't it?

Reply:
Nice !. Good thought, previously I thought this change may reproduce the same DML twice. On a second, this logic is only to find the latest schema change. I assume this should fix this edge case. I will amend the pull request and test it later on. 👍